### ISSUE TITLE
mvn package fails

### ISSUE THREAD

**Issue by unimatrix27** [Libresonic/libresonic#206]
Hi,

I cant execute mvn package successfully as the REST API fails to build due to some issue with a class, but I dont unterstand the output fully so I dont know if the dependency is missing or something else is wrong with it. To my understanding mvn package should download missing dependencies itself. Any hints?

Output: http://pastebin.com/g82sFKjr

**Comment by biconou** [comment]
Are you confident with your Maven installation ?

**Comment by unimatrix27** [comment]
i never used it before but i just installed it with apt-get on my ubuntu 16.04 LTS. Is there a way to check it? I tried finding hints for hours but they are mostly about specific other problems with packages. 

However with your hint i switched from openjdk 9 to oracle java using update-alternatives and for some reason it works now. So I guess my openjdk thing might be broken.

**Comment by muff1nman** [comment]
Try the code at PR #205

**Comment by muff1nman** [comment]
Or probably better, try the stable branch.

**Comment by unimatrix27** [comment]
i tried both PR 205 and stable. With PR 205 I still get HTTP 500, but now including an error page with the entries from the log. With stable I get HTTP 500 without error page. 

I also realized I had Openjdk 9 installed and rolled back to Openjdk 8.

**Comment by muff1nman** [comment]
So the logs here are from stable, tomcat8, Java 8 and Ubuntu?

**Comment by muff1nman** [comment]
Also if you have the localhost log, it might contain useful information.

**Comment by unimatrix27** [comment]
cleaned everything, rebuilt using stable, openjdk 8, tomcat8:

Bash-Log: http://pastebin.com/5T7D8S1n
/var/log/tomcat8/catalina.out    http://paste.ubuntu.com/23747017/
/var/log/tomcat8/catalina.2017-01-05.log   http://paste.ubuntu.com/23747023/
 /var/log/tomcat8/localhost.2017-01-05.log   http://paste.ubuntu.com/23747030/

**Comment by unimatrix27** [comment]
Im worried about this line in libresonic.log: 
`[2017-01-05` 17:19:28,050] INFO SettingsService - Java: 9-internal, OS: Linux` should it not say something like 8-internal? I completely removed openjdk9  I think.

**Comment by muff1nman** [comment]
Ah, so the java version reported via the command line is not necessarily the same as the one tomcat uses. I believe there was another debian user with an issue similar who had to change the java home in /etc somewhere.

**Comment by muff1nman** [comment]
But it looks like libresonic at least started up, which seems to be an improvement from the previous logs

**Comment by muff1nman** [comment]
https://github.com/Libresonic/libresonic/issues/189#issuecomment-269096448

Perhaps see if you can do the same in /etc/default/tomcat8

**Comment by unimatrix27** [comment]
Thanks again for your fast help. Indeed there was a leftover package openjdk-9-jre-headless, and previously I only installed openjdk-9-jre. Removing the leftover package solved the problem and now also libresonic is showing the right version in the log, AND I can see the login page. Yeah! So after all it is a debian/ubuntu issue. I found out tomcat just tries to use the first best java if JAVA_HOME is not set in /etc/default/tomcat8, and it does not care about the setting you do with update-alternatives.
